src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,20 +215,22 @@ public static String formatTime12H(Date date) {
         return sdf.format(date);
     }
 
-    public static String formatDateTimeForComments(Date date) {
+    public static String formatDateTimeForComments(Date date, double sessionTimeZone) {
         if (date == null) {
             return "";
         }
         SimpleDateFormat sdf = null;
         Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
+        TimeZone timeZone = getTimeZoneFromDoubleOffset(sessionTimeZone);
+        c.setTimeZone(timeZone);
         c.setTime(date);
         if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
-            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm");
-            sdf.setTimeZone(SystemParams.TIME_ZONE);
-            return sdf.format(date) + " NOON UTC";
+            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm 'NOON' 'UTC'Z");
+            sdf.setTimeZone(timeZone);
+            return sdf.format(date);
         }
-        sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm a zzz");
-        sdf.setTimeZone(SystemParams.TIME_ZONE);
+        sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm a 'UTC'Z");
+        sdf.setTimeZone(timeZone);
         return sdf.format(date);
```

Comment:
We don't need to change this as `Calendar.HOUR_OF_DAY` stores time in `24-hour-clock`, so 12 here means 12 NOON only.